hail/src/test/scala/is/hail/expr/ir/IRSuite.scala

```diff
@@ -4458,6 +4460,13 @@ class IRSuite extends HailSuite {
       ApplyScanOp(FastSeq.empty, FastSeq(Cast(Ref("x", TInt32), TInt64)), sumSig),
     )
     testFreeVarsHelper(streamScanIR)
+
+    val aggIR = ApplyAggOp(FastSeq(), FastSeq(Ref("x", TInt32)), sumSig)
+    val scanIR = ApplyScanOp(FastSeq(), FastSeq(Ref("x", TInt32)), sumSig)
+    val aggScanIR = MakeTuple.ordered(FastSeq(aggIR, scanIR))
+    val aggLetIR = AggLet("foo", Ref("y", TInt32), aggScanIR, true)
+    val letIR = Let(FastSeq("bar" -> Ref("z", TInt32)), aggLetIR)
+    println(FreeVariables(letIR, true, true))
```

Comment:
Is this test complete? can you assert that the free variables of an expression match some expected set of names?

Reply:
Whoops, thanks, this was just me trying to debug. It doesn't actually test anything interesting. I can try to add a bit more test coverage though.